XS✔ ◾ Improvements to code review prefixes (#7736)
SSWConsulting/SSW.Rules.Content: rules/use-prefixes-to-improve-code-review-communication/rule.md

<!--endintro-->

When conducting code reviews in a collaborative environment, it is essential to maintain effective communication. Utilizing prefixes in comments, as suggested on [conventionalcomments.org](https://conventionalcomments.org), can significantly enhance the code review process. Prefixes help convey the intent and impact of a comment, making it easier for the author and other team members to understand how to address it.

> Let's have a look at the following examples:

Reviewer: ```suggestion
Let's have a look at the following example from Bob Northwind:
```

:::greybox

> Hi @bob-northwind

Reviewer: ```suggestion
@bob-northwind
```
This is an example Code Review comment from Bob Northwind, hence the clarification in the previous comment.

:::

:::greybox

> Hi @bob-northwind

Reviewer: ```suggestion
@bob-northwind
```
This is an example Code Review comment from Bob Northwind, hence the clarification in the previous comment.

> **Suggestion**: This code could be better optimized.

Reviewer: ```suggestion
**suggestion**: This code could be better optimized.
```
The prefixes are deliberately kept in lowercase

:::greybox
@bob-northwind

> **Issue**: We must address this security vulnerability before merging.

Reviewer: ```suggestion
**issue**: We must address this security vulnerability before merging.
```
The prefixes are deliberately kept in lowercase

| Prefix         | non-/blocking | Description                                                                                                                                                                 |
| -------------- | ------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | **Issue**      | blocking      | Issues point out specific problems with the code. These can affect users or happen behind the scenes.                                                                       |
> | **TODO**       | blocking      | TODOs are essential changes. If they're complex, create a new task or PBI for them.                                                                                         |
> | **Question**   | blocking      | Questions are suitable when you're uncertain about something's relevance. Ask for clarification or investigation for a quick resolution.                                    |
> | **Praise**     | non-blocking  | Praises highlight something positive. It's good to include at least one in each review. Avoid false praise as it can be harmful. Look for something genuinely praiseworthy. |
> | **Suggestion** | non-blocking  | Suggestions offer ways to make the code better. Be clear about what you're suggesting and why it's an improvement.                                                          |
> | **Thought**    | non-blocking  | Thoughts are ideas that came up during the review. They don't block progress, but they can lead to more focused initiatives and learning opportunities.                     |
> | **Nitpick**    | non-blocking  | Nitpicks are minor, preference-based suggestions.                                                                                                                           |

Reviewer: ```suggestion
| **issue**      | blocking      | Issues point out specific problems with the code. These can affect users or happen behind the scenes.                                                                       |
| **todo**       | blocking      | TODOs are essential changes. If they're complex, create a new task or PBI for them.                                                                                         |
| **question**   | blocking      | Questions are suitable when you're uncertain about something's relevance. Ask for clarification or investigation for a quick resolution.                                    |
| **praise**     | non-blocking  | Praises highlight something positive. It's good to include at least one in each review. Avoid false praise as it can be harmful. Look for something genuinely praiseworthy. |
| **suggestion** | non-blocking  | Suggestions offer ways to make the code better. Be clear about what you're suggesting and why it's an improvement.                                                          |
| **thought**    | non-blocking  | Thoughts are ideas that came up during the review. They don't block progress, but they can lead to more focused initiatives and learning opportunities.                     |
| **nitpick**    | non-blocking  | Nitpicks are minor, preference-based suggestions.                                                                                                                           |
```
The prefixes are deliberately kept in lowercase
